File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsModule.java
package fr.free.nrw.commons.contributions;

import javax.inject.Named;
import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import fr.free.nrw.commons.kvstore.JsonKvStore;

/**
 * The Dagger Module for contributions-related presenters and other dependencies
 */
@Module
public abstract class ContributionsModule {

    @Binds
    public abstract ContributionsContract.UserActionListener bindsContributionsPresenter(
        ContributionsPresenter presenter
    );

    @Provides
    static JsonKvStore providesApplicationKvStore(
        @Named("default_preferences") JsonKvStore kvStore
    ) {
        return kvStore;
    }
}

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsModule.kt
package fr.free.nrw.commons.contributions

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

/**
 * The Dagger Module for contributions-related presenters and other dependencies
 */
@Module
@InstallIn(ActivityComponent::class)
abstract class ContributionsModule {

    @Binds
    abstract fun bindsContributionsPresenter(
        presenter: ContributionsPresenter?
    ): ContributionsContract.UserActionListener?
}